### ISSUE TITLE
Queue subscription Not Recreated (for an existing queue subscription)

### ISSUE THREAD

**Issue by nhakim-407etr** [SolaceProducts/terraform-provider-solacebroker#76]
### Bug Description

When trying to recreate queues with different attribute values or without any changes using the Solace Terraform provider, the queues are recreated/overwritten successfully. However, the subscription is not. An error message is displayed stating that the subscription already exists.  While terraform plan doesn’t show any issues, the apply command is throwing an error stating that the subscription already exists.

### Expected Behavior

The provider should have the capability to overwrite the subscription, similar to how it can overwrite queues.

### Steps to Reproduce

Define the queue and subscription resources(which already exist in the broker) in main.tf and terraform apply. 

### Solace Broker version

10.7.1.56-4

**Comment by nhakim-407etr** [comment]
![Capture](https://github.com/user-attachments/assets/8b6cd69a-9c4c-4a6a-b35e-1645d33dedcf)
screen shot of the error

**Comment by bczoma** [comment]
Hi @nhakim-407etr can you share a bit more to help reproduce?

Instead of disclosing your project is it possible to demonstrate the issue using this simple config?

```
resource "solacebroker_msg_vpn" "test" {
  msg_vpn_name = "test"
}

resource "solacebroker_msg_vpn_queue" "q1" {
  msg_vpn_name    = solacebroker_msg_vpn.test.msg_vpn_name
  queue_name      = "q1"
  egress_enabled  = true
}

resource "solacebroker_msg_vpn_queue_subscription" "foo" {
  msg_vpn_name       = solacebroker_msg_vpn_queue.q1.msg_vpn_name
  queue_name         = solacebroker_msg_vpn_queue.q1.queue_name
  subscription_topic = "foo/bar"
}
```

**Comment by nhakim-407etr** [comment]
Hello @bczoma, here's the configuration. You can either manually set up the queues and subscriptions or run the Terraform files and then remove the statefile. Subsequently, without modifying the main.tf, execute 'terraform plan' and 'apply' again. It will indicate two additions. The queue is established upon application, but the subscription doesn't succeed. I intend to use Terraform to manage the queues that were created manually or new ones. The queue is created irrespective of any attribute changes, but the subscription resource fails.


`resource "solacebroker_msg_vpn_queue" "q1" {
  msg_vpn_name                                   = var.msg_vpn_name
  queue_name                                     = "q1"
}

resource "solacebroker_msg_vpn_queue_subscription" "foo" {
  msg_vpn_name        = solacebroker_msg_vpn_queue.q1.msg_vpn_name
  queue_name          = solacebroker_msg_vpn_queue.q1.queue_name
  subscription_topic  = "foo/bar"
}
`


Error;
`Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # solacebroker_msg_vpn_queue.q1 will be created
  + resource "solacebroker_msg_vpn_queue" "q1" {
      + msg_vpn_name = "solace-407etr-dev-1"
      + queue_name   = "q1"
    }

  # solacebroker_msg_vpn_queue_subscription.foo will be created
  + resource "solacebroker_msg_vpn_queue_subscription" "foo" {
      + msg_vpn_name       = "solace-407etr-dev-1"
      + queue_name         = "q1"
      + subscription_topic = "foo/bar"
    }

Plan: 2 to add, 0 to change, 0 to destroy.
solacebroker_msg_vpn_queue.q1: Creating...
solacebroker_msg_vpn_queue.q1: Creation complete after 0s
solacebroker_msg_vpn_queue_subscription.foo: Creating...
╷
│ Error: SEMP call failed
│
│   with solacebroker_msg_vpn_queue_subscription.foo,
│   on main.tf line 23, in resource "solacebroker_msg_vpn_queue_subscription" "foo":
│   23: resource "solacebroker_msg_vpn_queue_subscription" "foo" {
│
│ request failed for https://<host>:943/SEMP/v2/config/msgVpns/<vpn_name>/queues/q1/subscriptions using POST, Problem with POST:
│ Subscription foo/bar already exists., ALREADY_EXISTS"`

**Comment by bczoma** [comment]
Hi @nhakim-407etr , some resources including "queue subscriptions" only support create-only behavior and cannot adjust an existing resource, it will fail if a resource already exists. Others, including "queues" can update existing resources and that is the default behavior of the Solace provider, if supported. To look up if this is supported, refer to the SEMP API documentation, `queue` for example (https://docs.solace.com/API-Developer-Online-Ref-Documentation/swagger-ui/software-broker/config/index.html#/queue). You can see that `queue` support the PATCH method for object update, where `subscriptions` doesn't.

Strictly, your use case should always start with "import" if you are adding resources to your state that were created outside the Terraform sessions related to the current state file. Terraform is not meant to automatically take ownership of resources it encounters. Using import shall avoid the behavior you are running into.

The misleading behavior of the Solace provider is rather that it "works" for queues that were created outside Terraform, than that it doesn't work for "queue subscriptions". A note that if the state file doesn't exist then 'terraform plan' will not show the issue as the POST method will only fail on the queue subscription as part of 'apply'.

Thanks for raising, the documentation will be updated to better describe this.

**Comment by nhakim-407etr** [comment]
Thank you @bczoma. I will terraform import and then then terraform apply for the scenario.

**Comment by bczoma** [comment]
Fixed in latest v1.1.0 release, added a note to https://registry.terraform.io/providers/SolaceProducts/solacebroker/latest/docs/guides/provider#importing-resources
